concordium-std-derive/src/lib.rs

```diff
@@ -1210,10 +1254,48 @@ fn schema_type_field_type(field: &syn::Field) -> syn::Result<proc_macro2::TokenS
 fn schema_type_fields(fields: &syn::Fields) -> syn::Result<proc_macro2::TokenStream> {
     match fields {
         syn::Fields::Named(_) => {
+            let mut used_field_names = HashMap::new();
             let fields_tokens: Vec<_> = fields
                 .iter()
                 .map(|field| {
-                    let field_name = field.ident.clone().unwrap().to_string(); // safe since named fields
+                    let (field_name, field_span) = if let Some(renamed_field_name) =
+                        find_field_attribute_value(&field.attrs, "rename")?
+                    {
+                        match renamed_field_name {
+                            syn::Lit::Str(renamed_field_name) => {
+                                (renamed_field_name.value(), renamed_field_name.span())
+                            }
+                            _ => {
+                                return Err(syn::Error::new(
+                                    renamed_field_name.span(),
+                                    "Rename attribute value must be a string.",
+                                ))
+                            }
+                        }
```

Comment:
I believe this part finding the rename attribute and checking whether it is the right type could be factored into a helper function, so that it is not duplicated here and in the `type_derive_worker` function.